coconut-rs/src/traits.rs

```diff
@@ -0,0 +1,17 @@
+pub trait Bytable {
+    fn to_byte_vec(&self) -> Vec<u8>;
+
+    fn from_byte_slice(slice: &[u8]) -> Self;
+}
+
+pub trait Base58
+where
+    Self: Bytable + Sized,
+{
+    fn from_bs58(x: &str) -> Self {
```

Comment:
I'd change the x type to be whatever `bs58::decode` takes as an argument, i.e. `I: AsRef<[u8]>`

Reply:
Unless generics are annoying in traits, which might very likely be the case

Reply:
Good idea, I'll give it a go